src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java

```diff
@@ -98,4 +98,23 @@ public ResponseEntity<Object> deleteApplication(@PathVariable("acsp_application_
         return acspService.deleteAcspApplication(acspId);
     }
 
+    @DeleteMapping("/transactions/{" + TRANSACTION_ID_KEY + "}/authorised-corporate-service-provider-applications/{acsp_application_id}")
+    public ResponseEntity<Object> deleteApplicationInfo(@PathVariable("acsp_application_id") String acspId,
+                                                        @PathVariable(value = TRANSACTION_ID_KEY) String transaction_id,
+                                                        @RequestAttribute(value = TRANSACTION_KEY) Transaction transaction) {
+        LOGGER.info("Received request to delete application for id: " + acspId);
+
+        if (transaction == null || !transaction.getId().equals(transaction_id)) {
```

Comment:
I understand the need of this but I dont think this condition will ever become true. 
There is an interceptor which will intercept the request before it reaches the controller get the transaction from the transaction id and add it as request Attribute. So transaction id and transaction will always match.